src/puya/awst/serialize.py

```diff
@@ -86,4 +88,7 @@ def awst_to_json(awst: nodes.AWST) -> str:
 
 
 def awst_from_json(json: str) -> nodes.AWST:
-    return _get_converter().loads(json, nodes.AWST)  # type: ignore[type-abstract]
+    try:
+        return _get_converter().loads(json, nodes.AWST)  # type: ignore[type-abstract]
+    except (ClassValidationError, IterableValidationError, BaseException) as err:
+        raise InternalError("Deserialization error: \n" + "\n".join(transform_error(err))) from err
```

Comment:
Since there's no source location, might as well make this a `ValueError`.

Reply:
changed to value error